components/google_my_business/app/google_my_business.app.ts

```diff
@@ -1,13 +1,191 @@
+    },
+    location: {
+      type: "string",
+      label: "Location",
+      description: "The location whose local posts will be listed. [See the documentation](https://developers.google.com/my-business/content/location-data#filter_results_when_you_list_locations) on how to filter locations.",
+      useQuery: true,
+      async options({
+        account, query,
+      }: Record<string, string>) {
+        const filter = query
+          ? (query.match(/[=:]/)
+            ? query
+            : `title="${query}"`).replace(/ /g, "+").replace(/"/g, "%22")
+          : undefined;
+
+        const locations: Location[] = await this.listLocations({
+          account,
+          filter,
+        });
+        return locations?.map?.(({
+          name, title,
+        }: Location) => ({
+          label: title,
+          value: this.getCleanName(name) as string,
+        })) ?? [];
+      },
+    },
+    review: {
+      type: "string",
+      label: "Review",
+      description: "Select a **Review** or provide a custom *Review Name*.",
+      async options({
+        account, location,
+      }: Record<string, string>) {
+        const reviews: Review[] = await this.listReviews({
+          account,
+          location,
+        });
+        return reviews?.map?.(({
+          name, title,
+        }: Location) => ({
+          label: title,
+          value: this.getCleanName(name) as string,
+        }));
+      },
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    getCleanName(name: string) {
+      return name?.split("/").pop();
+    },
+    _getHeaders() {
+      return {
+        Authorization: `Bearer ${this.$auth.oauth_access_token}`,
+      };
+    },
+    async _httpRequest({
+      $ = this,
+      ...args
+    }: HttpRequestParams): Promise<object> {
+      return axios($, {
+        ...args,
+        headers: this._getHeaders(),
+      });
+    },
+    async _paginatedRequest({
+      maxResults = 100,
+      maxPerPage = 100,
+      params,
+      resourceName,
+      ...args
+    }: PaginatedRequestParams): Promise<object[]> {
+      const result = [];
+      let pageToken: string;
+      let resultCount = 0;
+
+      do {
+        const pageSize = Math.min(maxResults - resultCount, maxPerPage);
+
+        const {
+          [resourceName]: resources,
+          nextPageToken,
+        } = await this._httpRequest({
+          params: {
+            ...params,
+            pageSize,
+            pageToken,
+          },
+          ...args,
+        });
+
+        if (resources) result.push(...resources);
+        pageToken = nextPageToken;
+        resultCount += pageSize;
+      } while (pageToken && resultCount < maxResults);
+
+      return result;
+    },
+    async listAccounts(): Promise<Account[]> {
+      const response = await this._httpRequest({
+        url: "https://mybusinessaccountmanagement.googleapis.com/v1/accounts",
```

Comment:
I initially did not set them because this app uses 3 different Google APIs - so we'd need 3 different base URL variables. Do you think we should do that, naming one variable for each?